File: android/src/main/java/ru/vvdev/yamap/utils/RouteManager.java
package ru.vvdev.yamap.utils;

import com.yandex.mapkit.map.MapObject;
import com.yandex.mapkit.transport.masstransit.Route;

import java.util.ArrayList;
import java.util.HashMap;

// todo: не используется
public class RouteManager {
    private HashMap<String, Route> data = new HashMap<>();
    private HashMap<String, ArrayList<MapObject>> mapObjects = new HashMap<>();

    public static String generateId() {
        return java.util.UUID.randomUUID().toString();
    }

    public void saveRoute(Route route, String id) {
        data.put(id, route);
    }

    public void putRouteMapObjects(String id, ArrayList<MapObject> objects) {
        mapObjects.put(id, objects);
    }

    public Route getRoute(String id) {
        return data.get(id);
    }

    public ArrayList<MapObject> getRouteMapObjects(String id) {
        return mapObjects.get(id);
    }
}

File: android/src/main/java/ru/vvdev/yamap/utils/RouteManager.kt
package ru.vvdev.yamap.utils

import com.yandex.mapkit.map.MapObject
import com.yandex.mapkit.transport.masstransit.Route
import java.util.UUID

// todo: не используется
class RouteManager {
    private val data = HashMap<String, Route>()
    private val mapObjects = HashMap<String, ArrayList<MapObject>>()

    fun saveRoute(route: Route, id: String) {
        data[id] = route
    }

    fun putRouteMapObjects(id: String, objects: ArrayList<MapObject>) {
        mapObjects[id] = objects
    }

    fun getRoute(id: String): Route? {
        return data[id]
    }

    fun getRouteMapObjects(id: String): ArrayList<MapObject> {
        return mapObjects[id]!!
    }

    companion object {
        fun generateId(): String {
            return UUID.randomUUID().toString()
        }
    }
}
